### ISSUE TITLE
OSX builds are bundling some libraries

### ISSUE THREAD

**Issue by jaimergp** [conda-forge/ambertools-feedstock#22]
Trying to update `perses` so it depends on `ambertools` instead of Omnia's `ambermini` results in an error due to clobbing:

```
ClobberError: This transaction has incompatible packages due to a shared path.
  packages: conda-forge/osx-64::libiconv-1.15-h01d97ff_1005, conda-forge/osx-64::ambertools-19.11-py37h8d9041b_1
  path: 'lib/libiconv.2.dylib'
```

The environment file used to trigger this behavior is listed here:

<details>

<summary>Environment</summary>

```
name: perses
channels:
  - conda-forge
  - openeye
  - omnia
  - defaults
​
dependencies:
  - pip
  - python
  - setuptools
  - numpy >=1.14
  - scipy
  - pymbar
  - openmm
  - openmoltools >=0.8.4
  - openmmtools
  - numba
  - netcdf4
  - matplotlib
  - seaborn
  - mdtraj
  - pdbfixer
  - lxml
  - networkx >=2.0
  - openeye-toolkits
  - dask
  - distributed
  - yank
  - progressbar2
  - tqdm
```

</details>

The CMake build system has a step after build where it bundles some libraries in `$PREFIX` for some reason (see [these lines](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=114533&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=0a2070c1-8027-59fc-ddea-6aec050600c1&l=15404)). This might be pulling too many files that are foreign to `ambertools`. This will need further investigation, I guess, since this step is not needed in Linux.

**Comment by jaimergp** [comment]
To provide more details, I think this is the complete list of dependencies manually bundled in that step:

```
 /usr/lib/libc++.1.dylib
 /usr/lib/libc++abi.dylib
 /usr/lib/libiconv.2.dylib
 /usr/lib/libncurses.5.4.dylib
 /usr/lib/libobjc.A.dylib
 /usr/lib/libresolv.9.dylib
 /usr/lib/libz.1.dylib
 /usr/local/miniconda/lib/libbz2.dylib
 /usr/local/miniconda/lib/libc++.1.0.dylib
 /usr/local/miniconda/lib/libc++.1.dylib
 /usr/local/miniconda/lib/libcrypto.1.1.dylib
 /usr/local/miniconda/lib/libicudata.64.2.dylib
 /usr/local/miniconda/lib/libicudata.64.dylib
 /usr/local/miniconda/lib/libicui18n.64.2.dylib
 /usr/local/miniconda/lib/libicui18n.64.dylib
 /usr/local/miniconda/lib/libicuio.64.dylib
 /usr/local/miniconda/lib/libicutu.64.2.dylib
 /usr/local/miniconda/lib/libicutu.64.dylib
 /usr/local/miniconda/lib/libicuuc.64.2.dylib
 /usr/local/miniconda/lib/libicuuc.64.dylib
 /usr/local/miniconda/lib/liblzma.5.dylib
 /usr/local/miniconda/lib/libncurses.6.dylib
 /usr/local/miniconda/lib/libncursesw.6.dylib
 /usr/local/miniconda/lib/libreadline.8.dylib
 /usr/local/miniconda/lib/libssl.1.1.dylib
 /usr/local/miniconda/lib/libtcl8.6.dylib
 /usr/local/miniconda/lib/libtinfo.6.dylib
 /usr/local/miniconda/lib/libtinfow.6.dylib
 /usr/local/miniconda/lib/libtk8.6.dylib
 /usr/local/miniconda/lib/libz.1.2.11.dylib
 /usr/local/miniconda/lib/libz.1.dylib
 $BUILD_PREFIX/lib/libgcc_s.1.dylib
 $PREFIX//lib/libc++abi.dylib
 $PREFIX//lib/libncurses.5.4.dylib
 $PREFIX//lib/libresolv.9.dylib
```

This is no good... I can see libraries from the system and the root miniconda installation! Hopefully disabling this step in PR #23 is enough.

**Comment by jaimergp** [comment]
I have seen yet another problem in my local OSX tests that were not caught in the CI testing phase:

```
$ conda create -n ambertools-cf -c conda-forge ambertools
$ conda activate ambertools-cf
$ python -c "import parmed"
```

(Sorry can't copy from the VM)

![image](https://user-images.githubusercontent.com/2559438/73348129-206a1f80-4289-11ea-86c9-2c5d9a7bcaa4.png)

I can't reproduce this error in Linux, so I guess it's due to the bundling process.

**Comment by jchodera** [comment]
@jaimergp: This worked without incident on my `osx` installation:
```
$ conda create -n ambertools-cf -c conda-forge ambertools
$ conda activate ambertools-cf
$ python -c "import parmed"
```

**Comment by jaimergp** [comment]
Oh, interesting. I will have to nuke my Miniconda installation, I guess 😬

**Comment by SimonBoothroyd** [comment]
I think relatedly - it seems the OSX version of `ambertools` does not play nicely with `rdkit`:

```
conda create --name ambertest
conda activate ambertest
conda install --yes -c conda-forge ambertools rdkit
python -c "from rdkit.Chem import AllChem"
```
yields
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/boothros/miniconda3/envs/ambertest/lib/python3.8/site-packages/rdkit/Chem/__init__.py", line 24, in <module>
    from rdkit.Chem.rdmolops import *
ImportError: dlopen(/Users/boothros/miniconda3/envs/ambertest/lib/python3.8/site-packages/rdkit/Chem/rdmolops.so, 2): Library not loaded: @rpath/libiconv.2.dylib
  Referenced from: /Users/boothros/miniconda3/envs/ambertest/lib/libxml2.2.dylib
  Reason: Incompatible library version: libxml2.2.dylib requires version 9.0.0 or later, but libiconv.2.dylib provides version 7.0.0
```

**Comment by j-wags** [comment]
@SimonBoothroyd's case reproduces for me.

**Comment by jaimergp** [comment]
Hopefully fixed by #23. I'll leave this open some more days in case somebody wants to report something else, but according to @SimonBoothroyd, it seems to work now.

**Comment by jaimergp** [comment]
Closing as no issues have been reported in 3 weeks.
